src/Java.Interop.Tools.JavaCallableWrappers/Java.Interop.Tools.JavaCallableWrappers/CecilExtensions.cs

```diff
@@ -45,7 +45,7 @@ internal static ExportFieldAttribute ToExportFieldAttribute (CustomAttribute att
 		while ((bmethod = method.GetBaseDefinition (cache)) != method) {
 			method = bmethod;
 
-			if (method.AnyCustomAttributes (typeof (RegisterAttribute))) {
+			if (GetMethodRegistrationAttributes (method).Any ()) {
```

Comment:
Is this called by xamarin-android at all?

Just want to make sure we wouldn't make this too much worse:
* https://github.com/xamarin/xamarin-android/issues/8920

I guess this could create an attribute instance, an `IEnumerable`, and then call `.Any()`.

Reply:
Yes this codepath *is* used by .NET for Android, in which case the problem with this change is that it could introduce additional object allocations that aren't needed.  I'll look into updating this.